### DISCUSSION TITLE
Token lifespan constraint to be greater than 15 minutes

### DISCUSSION THREAD

**Discussion by betweenstrap21** [minio/minio#12572] [Q&A] [answered]
Hello,

When using STS, the temporary credentials returned can be configured but cannot be less than 15 minutes. The consequence is that the Access Token received from an IDP cannot have a lifespan of less than 15 Minutes (900 seconds). 
The corresponding lines where that constraint is coded are in the following file: 
https://github.com/minio/minio/blob/d40268d8991516553816afbd8218674661eadfae/internal/config/identity/openid/jwt.go#L134

The problem is that the Security rules in my company are strict and an Access token cannot have a lifespan of more than 5 minutes. As a consequence I cannot use the STS service. How was this value of 15 minutes chosen ? Is there any possibility that this value is changed in min.io ?

Thank you

Regards

**Comment by harshavardhana** [comment, marked as answer]
@betweenstrap21 our implementation complies with AWS S3 STS API - changing this will require breaking compatibility.

**Reply by betweenstrap21** [reply]
Thank you for your answer. I do not have any problem with the STS credentials having a minimum of 15 minutes lifespan which is indeed the S3 standard. However why is there the requirement that the incoming access token has a lifespan superior to the STS credentials : I understand this is a security precaution on min.io side (min.io does not want to issue sts credentials that could last more than the original access token) but this should be up to the person configuring min.io to make such a decision.  In my case I can take the risk to have an Access Token with a 5 minutes lifespan while having STS credentials with a 15 minutes lifespan as long as my Access Token is valid at the time I make the initial request. 
As a comparison, looking at the Redhat Ceph STS code, if I do not mistake, I do not see the same constraint while they are also implementing standard S3 STS.
